lms/djangoapps/support/serializers.py

```diff
@@ -84,17 +84,40 @@ def serialize_user_info(user, user_social_auths=None):
     return user_info
 
 
-def serialize_sso_records(user_social_auths):
+def serialize_sso_records(user_social_auths, user_social_auths_history):
     """
     Serialize user social auth model object
     """
     sso_records = []
     for user_social_auth in user_social_auths:
+        history = serialize_sso_history(
+            user_social_auth,
+            user_social_auths_history
+        )
         sso_records.append({
             'provider': user_social_auth.provider,
             'uid': user_social_auth.uid,
             'created': user_social_auth.created,
             'modified': user_social_auth.modified,
             'extraData': json.dumps(user_social_auth.extra_data),
+            'history': history,
         })
     return sso_records
+
+
+def serialize_sso_history(user_social_auth, user_social_auths_history):
+    """
+    Serialize history for user social auth model object
+    """
+    history = []
+    for sso_history in user_social_auths_history:
+        if sso_history.provider == user_social_auth.provider:
```

Comment:
Can there be more than one provider and associated histories? if yes, how will they get handle?

Reply:
Right now, each SSO is associated with one provider and can have multiple histories. Hence, each provider is associated with multiple histories which are returned in the form of an array.

Reply:
What happens if the user has multiple SSO records with the same provider? Wouldn't this results in each SSO record in the response having a `history` describing both instances?